markdown/htmlparser.py

```diff
@@ -264,6 +272,14 @@ def parse_starttag(self, i):  # pragma: no cover
         if end.endswith('/>'):
             # XHTML-style empty tag: <span attr="value" />
             self.handle_startendtag(tag, attrs)
+        elif tag in self.empty_tags:
```

Comment:
There should be no changes to the `parse_starttag` method. We only override this method to fix the upstream issue. In all other respects this is an exact copy of the parent class. See the comment on this method would suggest the same. We need to address the issue in the public functions first if we can.